### PR TITLE
CVSL-1849 Fixing nav and adding validation when creating hard stop licences

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: server/routes/creatingLicences/types/hardStopLicenceToSubmit.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  @Expose()
+  @IsNotEmpty({ message: "Select 'Change' to go back and add appointment telephone number" })
+  appointmentContact: string
+
+  @Expose()
+  @IsNotEmpty({ message: "Select 'Change' to go back and add appointment date and time" })
+  appointmentTimeType: AppointmentTimeType
+}
+
+export default HardStopLicenceToSubmit
```
Comment (andrewrlee) [inline server/routes/creatingLicences/types/hardStopLicenceToSubmit.ts:23]: new validation form (differs due to no details about conditions)

File: server/routes/initialAppointment/handlers/hardStop/index.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ export default function Index({ licenceService, conditionService, communityServi
     post('(/create|/edit)/id/:licenceId/initial-meeting-time', controller.POST, DateTime)
   }
   {
-    const controller = new ViewAndPrintLicenceRoutes(licenceService, communityService)
+    const controller = new ViewAndPrintLicenceRoutes(licenceService)
```
Comment (andrewrlee) [inline server/routes/initialAppointment/handlers/hardStop/index.ts:68]: unused dependency

File: server/routes/viewingLicences/handlers/viewCases.ts [hunk 3]
```diff
@@ -59,7 +63,10 @@ export default class ViewAndPrintCaseRoutes {
         licence.versionOf && licence.status === LicenceStatus.SUBMITTED
           ? `?lastApprovedVersion=${licence.versionOf}`
           : ''
-      return `/licence/view/id/${licence.id}/show${query}`
+
+      return cvlFields.isInHardStopPeriod && this.isEditableInHardStop(licence)
+        ? `/licence/hard-stop/id/${licence.id}/check-your-answers${query}`
+        : `/licence/view/id/${licence.id}/show${query}`
```
Comment (andrewrlee) [inline server/routes/viewingLicences/handlers/viewCases.ts:69]: got to hardstop specific check-your-answers page if the licence exists

File: server/routes/viewingLicences/handlers/viewLicence.ts [hunk 4]
```diff
@@ -75,7 +75,13 @@ export default class ViewAndPrintLicenceRoutes {
...
   POST = async (req: Request, res: Response): Promise<void> => {
     const { licenceId } = req.params
-    const { user } = res.locals
+    const { user, licence } = res.locals
+
+    const errors = await this.validateLicence(licence)
+    if (errors.length > 0) {
+      req.flash('validationErrors', JSON.stringify(errors))
+      return res.redirect('back')
+    }
```
Comment (andrewrlee) [inline server/routes/viewingLicences/handlers/viewLicence.ts:84]: validate on submission

File: server/views/partials/hardStop/inductionMeetingDetailsSummary.njk [hunk 5]
```diff
@@ -28,7 +28,7 @@
                             href: routepath + "/initial-meeting-name",
                             text: "Change",
                             visuallyHiddenText: "name",
-                            attributes: { "data-qa": "person-change-link" }
+                            attributes: { "id": "appointmentPersonType", "data-qa": "person-change-link" }
```
Comment (andrewrlee) [inline server/views/partials/hardStop/inductionMeetingDetailsSummary.njk:31]: Adding links so when you click an error the nav highlights the correct link

**Review by nishanthmahasamudram — APPROVED** [review]
LGTM

**Pull request merged**